isobus/src/isobus_heartbeat.cpp

```diff
@@ -0,0 +1,287 @@
+//================================================================================================
+/// @file isobus_heartbeat.cpp
+///
+/// @brief Implements an interface for sending and receiving ISOBUS heartbeats.
+/// The heartbeat message is used to determine the integrity of the communication of messages and
+/// parameters being transmitted by a control function. There may be multiple instances of the
+/// heartbeat message on the network, and CFs are required transmit the message on request.
+/// As long as the heartbeat message is transmitted at the regular
+/// time interval and the sequence number increases through the valid range, then the
+/// heartbeat message indicates that the data source CF is operational and provides
+/// correct data in all its messages
+///
+/// @author Adrian Del Grosso
+///
+/// @copyright 2024 The Open-Agriculture Developers
+//================================================================================================
+#include "isobus/isobus/isobus_heartbeat.hpp"
+
+#include "isobus/isobus/can_general_parameter_group_numbers.hpp"
+#include "isobus/isobus/can_network_manager.hpp"
```

Comment:
thought: I think we have a circular dependency currently, correct me if I'm wrong. Maintaining (and testing) this interface should be easier if we can remove it. I'll leave a few more comments on how I would approach it

Reply:
Circular dependency now removed